sakuya/mewo.py

```diff
@@ -0,0 +1,123 @@
+from dataclasses import dataclass
+import random
+from typing import Dict
+
+from discord import Guild
+from discord.ext import commands
+from discord.ext.commands import Bot
+
+from .db import db, Guild as _Guild, Member as _Member
+
+
+random_mewo = [
+    "ME.. oh you're not mcmower.",
+    "MEWO",
+    "You didn't really expect me to say mewo did you"
+]
+
+
+@dataclass
+class GuildState:
+    guild: Guild
+    mewo_enabled: bool
+
+
+class Mewo(commands.Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.guilds: Dict[Guild, GuildState] = dict()
+        self.data_loaded = False
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+        # This event fires on reconnects, but we only want it to run once
+        if not self.data_loaded:
+            self.data_loaded = True
+            self.load_from_db()
+            print('Mewo response ready.')
+
+    def load_from_db(self):
+        for g in db.query(_Guild).filter(_Guild.mewo_enabled.isnot(None)).all():
+            guild: Guild = self.bot.get_guild(g.id)
+            if guild:
+                if g.mewo_enabled:
+                    self.guilds[guild] = GuildState(guild=guild, mewo_enabled=True)
+                else:
+                    print(f"Mewo is not enabled on {guild.name}! Mewo responses disabled in guild.")
+            else:
+                print(f"Guild {g.id} not found during Mewo init.")
+
+    @commands.command()
+    async def mewo(self, ctx):
+        state = self.guilds.get(ctx.guild)
+        if state:
+
+            memberid = ctx.message.mentions[0].id if len(ctx.message.mentions) == 1 else ctx.author.id
+
+            member = db.query(_Member).get((memberid, ctx.guild.id)) or _Member(user_id=memberid,
+                                                                                guild_id=ctx.guild.id)
+
+            if member.mewo_text:
+                msg = member.mewo_text
+            else:
+                msg = random.choice(random_mewo)
+            
+            await ctx.send(msg)
+            return
+
+    @commands.command()
+    @commands.has_permissions(manage_nicknames=True)
+    async def mewoset(self, ctx, *, mewo_text):
+        state = self.guilds.get(ctx.guild)
+        if state:
+
+            if len(ctx.message.mentions) == 1:
+                memberid = ctx.message.mentions[0].id
+                print(mewo_text)
+                mewo_text = str(mewo_text.split(" ",1)[1])
+                pronoun = "their"
+            else:
+                memberid = ctx.author.id
+                pronoun = "your"
+
+            member = db.query(_Member).get((memberid, ctx.guild.id)) or _Member(user_id=memberid,
+                                                                                guild_id=ctx.guild.id)
+            previous_mewo = member.mewo_text
```

Comment:
nit: You could inline this variable as `member.mewo_text` isn't set before `previous_mewo` is read from, but having the explicit variable name is more readable...

@Trif4 What do you think?

Reply:
This one is also due to changes in code arrangement between testing; previously the member lookup and response were in a different order that neccessitated the use of an additional var, but that caused a different issue (hence the re-order). I do like this one for the readability aspect, but will leave @Trif4 to decide

Reply:
Personally I'd inline it; you can still preserve readability by just adding a comment like "User already has a mewo".